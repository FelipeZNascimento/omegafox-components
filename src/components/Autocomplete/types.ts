export type TDropdownItem = {
  id: number;
  name: string;
  abbreviationEn: string;
};

export interface IAutocompleteProps {
  dropdownList: TDropdownItem[];
  onSelect: (item: TDropdownItem) => void;
}
