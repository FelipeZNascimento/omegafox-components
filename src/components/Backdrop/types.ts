export interface IBackdropProps {
  align?: 'left' | 'center' | 'right';
  children: JSX.Element;
  isOpen: boolean;
}
