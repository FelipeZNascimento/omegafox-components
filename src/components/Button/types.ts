export interface IButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
  children?: React.ReactNode | null;
  icon?: React.ReactNode | null;
  isDisabled?: boolean;
  isDarkMode?: boolean;
  isSelected?: boolean;
  leftIcon?: boolean;
  size?: 'small' | 'medium' | 'large';
  shape?: 'rounded' | 'squared';
  shadowed?: boolean;
  tooltip?: string;
  variant?: 'primary' | 'danger' | 'neutral' | 'confirm' | 'warning';
  onClick: () => void;
}
