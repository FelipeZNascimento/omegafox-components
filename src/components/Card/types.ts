export interface ICardProps {
  isForceMobile?: boolean;
  isSelected: boolean;
  title: string;
  subtitle?: string | null;
  onClick?: null | (() => void);
  renderingStatusFunction?: null | (() => JSX.Element | null);
}
