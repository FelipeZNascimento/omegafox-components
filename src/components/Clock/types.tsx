import {
  FOOTBALL_MATCH_STATUS,
  AMERICAN_FOOTBALL_MATCH_STATUS
} from './constants';
import { TBetValues } from '../types';

type TClock = {
  time: number;
};

type FootballKeys = keyof typeof FOOTBALL_MATCH_STATUS;
export type TClockFootball = TClock & {
  status: typeof FOOTBALL_MATCH_STATUS[FootballKeys];
};

type AmericanFootballKeys = keyof typeof AMERICAN_FOOTBALL_MATCH_STATUS;
export type TClockAmericanFootball = TClock & {
  status: typeof AMERICAN_FOOTBALL_MATCH_STATUS[AmericanFootballKeys];
};

export interface IClockProps {
  betValue: TBetValues | null;
  clock: TClockAmericanFootball | TClockFootball;
  isMatchStarted?: boolean;
  isExpandable: boolean;
  isExpanded: boolean;
  isMatchEditable: boolean;
  timestamp: number;
}

export interface IClockMobileProps {
  clock: TClockAmericanFootball | TClockFootball;
  timestamp: number;
}
