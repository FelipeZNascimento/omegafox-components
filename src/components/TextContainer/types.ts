import { TBorderPosition } from 'components/types';

export interface ITextContainerProps {
  backgroundImage?: string;
  borderPosition?: TBorderPosition;
  children: JSX.Element;
}
