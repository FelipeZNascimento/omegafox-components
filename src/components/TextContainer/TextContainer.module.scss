@use "../../styling/colors.scss" as colors;
@use "../../styling/color_utils.scss" as color_utils;

@import "../../styling/alignments.scss";
@import "../../styling/font-sizes.scss";
@import "../../styling/margins.scss";
@import "../../styling/paddings.scss";

.container {
  @include color_utils.shadowed(colors.$black, 0.3);
  @include padded(all, m);
  @include flex-aligned(flex-start, flex-start);

  background-color: colors.$grey1;
  background-size: contain;
  background-position: center;
  background-repeat: no-repeat;
  background-origin: content-box;
  border-radius: $border-radius;
  flex-direction: column;
  flex: 1;
  overflow-y: auto;

  &BottomLeft {
    border-bottom-left-radius: 0;
  }
  &BottomRight {
    border-bottom-right-radius: 0;
  }
  &TopRight {
    border-top-right-radius: 0;
  }
  &TopLeft {
    border-top-left-radius: 0;
  }
}
