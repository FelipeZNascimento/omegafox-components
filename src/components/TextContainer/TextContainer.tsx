import React from 'react';

import classNames from 'classnames';
import { ITextContainerProps } from './types';
import styles from './TextContainer.module.scss';

export const TextContainer = ({
  backgroundImage,
  borderPosition = 'bottomLeft',
  children
}: ITextContainerProps) => {
  const containerClass = classNames(styles.container, {
    [styles.containerBottomLeft]: borderPosition === 'bottomLeft',
    [styles.containerBottomRight]: borderPosition === 'bottomRight',
    [styles.containerTopLeft]: borderPosition === 'topLeft',
    [styles.containerTopRight]: borderPosition === 'topRight'
  });

  return (
    <div
      className={containerClass}
      style={{
        backgroundImage: `url(${backgroundImage})`
      }}
    >
      {children}
    </div>
  );
};
