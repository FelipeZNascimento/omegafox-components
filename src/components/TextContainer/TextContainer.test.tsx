import React from 'react';
import { render } from '@testing-library/react';
import { TextContainer } from './TextContainer';
import { ITextContainerProps } from './types';

describe('<TextContainer />', () => {
  const renderComponent = ({ children }: ITextContainerProps) =>
    render(<TextContainer>{children}</TextContainer>);

  it('should render', () => {
    const { container } = renderComponent({ children: <p>Texto</p> });

    expect(container).toMatchSnapshot();
  });
});
