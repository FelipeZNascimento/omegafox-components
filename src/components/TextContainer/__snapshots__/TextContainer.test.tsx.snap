// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextContainer /> should render 1`] = `
<div>
  <div
    class="container containerBottomLeft"
    style="background-image: url(undefined);"
  >
    <p>
      Texto
    </p>
  </div>
</div>
`;
